### ISSUE TITLE
6.16.1 Build failed in cloud build ios

### ISSUE THREAD

**Issue by Thaina** [firebase/quickstart-unity#836]
### [REQUIRED] Please fill in the following fields:

  * Unity editor version: 2020.1.9f1
  * Firebase Unity SDK version: 6.16.1
  * Source you installed the SDK: UPM
  * Problematic Firebase Component: Analytics maybe?
  * Other Firebase Components in use: Auth, FireStore, CrashLytics
  * Additional SDKs you are using: Facebook, AdMob, OneSignal, UnityPurchasing
  * Platform you are using the Unity editor on: Windows + Cloud Build
  * Platform you are targeting: iOS
  * Scripting Runtime: IL2CPP

#### Steps to reproduce:

Just upgrade every firebase I used to 6.16.1 and build on unity cloud build

Have you been able to reproduce this issue with just the Firebase Unity quickstarts (this GitHub project)?
- Can't test that on cloud build

What's the issue repro rate? (eg 100%, 1/5 etc)
- 100%

What happened? How can we make the problem occur?
- I think the place that related to Build fail is here, at 2525

```
2519: [Unity] The domain/default pair of (com.apple.dt.Xcode.plist, IDECustomDerivedDataLocation) does not exist
2520: [Unity] ERROR: iOS framework addition failed due to a CocoaPods installation failure. This will will likely result in an non-functional Xcode project.
2521: [Unity] After the failure, "pod repo update" was executed and succeeded. "pod install" was then attempted again, and still failed. This may be due to a broken CocoaPods installation. See: https://guides.cocoapods.org/using/troubleshooting.html for potential solutions.
2522: [Unity] pod install output:
2523: [Unity] Analyzing dependencies
2524: [Unity] Cloning spec repo `cocoapods` from `https://github.com/CocoaPods/Specs.git`
2525: [Unity] [!] CocoaPods could not find compatible versions for pod "GoogleUtilities/NSData+zlib":
2526: [Unity]   In Podfile:
2527: [Unity]     Firebase/Analytics (= 6.32.2) was resolved to 6.32.2, which depends on
2528: [Unity]       Firebase/Core (= 6.32.2) was resolved to 6.32.2, which depends on
2529: [Unity]         FirebaseAnalytics (= 6.8.2) was resolved to 6.8.2, which depends on
2530: [Unity]           GoogleUtilities/NSData+zlib (~> 6.7)
2531: [Unity]     GoogleUtilities (= 6.6.0) was resolved to 6.6.0, which depends on
2532: [Unity]       GoogleUtilities/NSData+zlib (= 6.6.0)
2533: [Unity] Specs satisfying the `GoogleUtilities/NSData+zlib (= 6.6.0), GoogleUtilities/NSData+zlib (~> 6.7)` dependency were found, but they required a higher minimum deployment target.
2534: [Unity] pod repo update output:
```

Currently now I would try reverting to 6.15.2 and see the result

**Comment by Thaina** [comment]
6.15.2 don't cause build failed. 6.16.1 is

**Comment by DellaBitta** [comment]
Hi @Thaina,

Hello again!  I couldn't reproduce this locally via Unity with the Firebase podfiles in question.  Two things that first come to mind:

  1. Could it be that you're targeting an iOS that's below the minimum supported version for Firebase?
  1.  It could be that one of the other assets that you're importing (Facebook, OneSignal, etc) uses a different version of GoogleUtilities compared to the version that the Firebase SDKs depend on.  

Is there more to the error log?  Are you willing to post your podfile?

Thanks!

**Comment by Thaina** [comment]
> Hi @Thaina,
> 
> Hello again! I couldn't reproduce this locally via Unity with the Firebase podfiles in question.  Two things that first come to mind:
> 
> 1. Could it be that you're targeting an iOS that's below the minimum supported version for Firebase?

Today I have try to build my project again with 6.16.1 and still got the same error

I use cloud build and this is the version of xcode I could found from log

```
25179:  + Installing cocoapods dependencies:
25180:        ! .xcworkspace file could not be found.
25181:        Analyzing dependencies
25182:        [!] CocoaPods could not find compatible versions for pod "GoogleUtilities/Environment":
25183:          In Podfile:
25184:            Firebase/Messaging (= 6.32.2) was resolved to 6.32.2, which depends on
25185:              FirebaseMessaging (~> 4.6.2) was resolved to 4.6.2, which depends on
25186:                GoogleUtilities/Environment (~> 6.7)
25187:        
25188:            GoogleUtilities (= 6.6.0) was resolved to 6.6.0, which depends on
25189:              GoogleUtilities/Environment (= 6.6.0)
25190:  + Building with Xcode 11.4.1
```

> 2. It could be that one of the other assets that you're importing (Facebook, OneSignal, etc) uses a different version of GoogleUtilities compared to the version that the Firebase SDKs depend on.

This is possible but how could I check that?

> Is there more to the error log?  Are you willing to post your podfile?

I am not sure I can expose my podfile so very sorry. But I have tried to find error in my build log

```
13127: [Unity] ERROR: iOS framework addition failed due to a CocoaPods installation failure. This will will likely result in an non-functional Xcode project.
13128: [Unity] After the failure, "pod repo update" was executed and succeeded. "pod install" was then attempted again, and still failed. This may be due to a broken CocoaPods installation. See: https://guides.cocoapods.org/using/troubleshooting.html for potential solutions.
13129: [Unity] pod install output:
13130: [Unity] Analyzing dependencies
13131: [Unity] Cloning spec repo `cocoapods` from `https://github.com/CocoaPods/Specs.git`
13132: [Unity] [!] CocoaPods could not find compatible versions for pod "GoogleUtilities/Environment":
13133: [Unity]   In Podfile:
13134: [Unity]     Firebase/Messaging (= 6.32.2) was resolved to 6.32.2, which depends on
13135: [Unity]       FirebaseMessaging (~> 4.6.2) was resolved to 4.6.2, which depends on
13136: [Unity]         GoogleUtilities/Environment (~> 6.7)
13137: [Unity]     GoogleUtilities (= 6.6.0) was resolved to 6.6.0, which depends on
13138: [Unity]       GoogleUtilities/Environment (= 6.6.0)
13139: [Unity] pod repo update output:
13140: [Unity] Updating spec repo `cocoapods`
13141: [Unity]   master
13142: [Unity]   HEAD is now at c55031349a7a [Add] QVEditorKit 1.0.38
13143: [Unity]   remote: Enumerating objects: 53, done.        
13144: [Unity]   remote: Counting objects:   1% (1/53)        
```

Seem like exactly the same as I posted at 15 days ago

The actual error that cause build failed is the one below. But I believe the above error is the cause of this below error

```
26226: ▸ Compiling GADUInterface.m
26227: ❌;  /BUILD_PATH/mycompany.new-unity-project.default-ios/temp20201106-3789-3dc8nz/Libraries/Plugins/iOS/GADUAdLoader.h:5:9: 'GoogleMobileAds/GoogleMobileAds.h' file not found
26228: #import <GoogleMobileAds/GoogleMobileAds.h>
26229: ▸ Compiling GADUMChartboostInterface.m
26230: ** ARCHIVE FAILED **
```

**Comment by patm1987** [comment]
Hi @Thaina ,

First, if you'd like you can get support via the [bug reporting tool](https://firebase.google.com/support/troubleshooter/report/bugs). In cases where you can't expose your Podfile, this is secure and non-public and the agents there have other debugging tools for your project itself if it becomes necessary that we simply can't access here.

Second, if you have access to a Mac, are you able to successfully build locally?

Third, can you try changing how Cocoapods integrates from Xcode Workspace to Xcode Project?
![image](https://user-images.githubusercontent.com/1111611/98425881-0fd6a900-2054-11eb-94db-146bb03e9baa.png)
This will be under "Assets>External Dependency Manager>iOS Resolver>Settings"

Finally, can you try doing a cloud build with any of the [quickstart apps](https://github.com/firebase/quickstart-unity)? I see Messaging appearing a lot so I'd try [that one](https://github.com/firebase/quickstart-unity/tree/master/messaging/testapp). We've done limited testing on Cloud Build, so verifying that it's an issue with our quickstart will help a ton!

Thanks,
--Patrick
